src/rust/engine/src/scheduler.rs

```diff
@@ -270,8 +281,19 @@ type Root = Select;
 
 pub type RootResult = Result<Value, Failure>;
 
-impl ContextFactory for Arc<Core> {
-  fn create(&self, entry_id: EntryId) -> Context {
-    Context::new(entry_id, self.clone())
+#[derive(Clone)]
+struct RootContext {
```

Comment:
What value is this providing over just using an `Arc<Core>`?

Reply:
Is this because the NodeContext trait comes from our new graph crate and we don't own Arc?

Reply:
@jsirois : Bingo. Coherence.